src/libs/actions/Report.js

```diff
@@ -804,6 +804,20 @@ function handleReportChanged(report) {
         return;
     }
 
+    // It is possible that we optimistically created a DM/group-DM for a set of users for which a report already exists.
+    // In this case, the API will let us know by returning a preexistingReportID.
+    // We should clear out the optimistically created report and re-route the user to the preexisting report.
+    if (report && report.reportID && report.preexistingReportID) {
+        Onyx.set(`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`, null);
+        const currentRoute = navigationRef.current && navigationRef.current.getCurrentRoute();
```

Comment:
Ah, you're right we could.

Reply:
Though `Navigation.getActiveRoute` returns a string like `/r/1195783708973664` which we'd have to copmare like this: 

`if (Navigation.getActiveRoute === `/r/${report.reportID}`

Which doesn't actually feel better to me. So I think I prefer it the way it is. What do you think @mananjadhav?

Reply:
Alternatively, I could create a `Navigation.getActiveRouteObject` function.

Reply:
Yeah that sounds good

Reply:
Hmm if we use `getActiveRoute` we could do:

```
if (Navigation.getActiveRoute().endsWith(`/${report.reportID}`) {
```

Which doesn't really seem that bad to me, I do feel like that looks better than the current implementation, especially since we're reusing a useful looking function :D

Also I see `navigationRef.current.getCurrentRoute().path is undefined in the first navigation.` in this comment: https://github.com/Expensify/App/blob/ecf0ec2a578a530e729e9465f825099f01dfde19/src/libs/Navigation/Navigation.js#L205-L206

Which kinda scares me :D

Reply:
Ahhh fair points. Ok, I'm down with that approach, will update.